js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorerSidebar.tsx

```diff
@@ -335,93 +490,105 @@ const Node = ({
-                        });
-                        onClick([node.assetKey], e, false);
-                      }}
-                    />
-                    {upstream.length || downstream.length ? <MenuDivider /> : null}
-                    {upstream.length ? (
-                      <MenuItem
-                        text="Select upstream"
-                        icon="panel_show_left"
-                        onClick={(e) => {
-                          e.stopPropagation();
-                          // TODO: Hook up selecting the nodes
-                          showUpstreamGraph();
-                        }}
-                      />
-                    ) : null}
-                    {downstream.length ? (
-                      <MenuItem
-                        text="Select downstream"
-                        icon="panel_show_right"
-                        onClick={(e) => {
-                          e.stopPropagation();
-                          // TODO: Hook up selecting the nodes
-                          showDownstreamGraph();
-                        }}
-                      />
-                    ) : null}
-                    {upstream.length || downstream.length ? <MenuDivider /> : null}
-                    {upstream.length ? (
-                      <MenuItem
-                        text="Show upstream graph"
-                        icon="arrow_back"
-                        onClick={(e) => {
-                          e.stopPropagation();
-                          showUpstreamGraph();
-                        }}
-                      />
-                    ) : null}
-                    {downstream.length ? (
-                      <MenuItem
-                        text="Show downstream graph"
-                        icon="arrow_forward"
-                        onClick={(e) => {
-                          e.stopPropagation();
-                          showDownstreamGraph();
-                        }}
-                      />
-                    ) : null}
-                  </Menu>
-                }
-                hoverOpenDelay={100}
-                hoverCloseDelay={100}
-                placement="right"
-                shouldReturnFocusOnClose
+              <div
+                style={{
+                  display: 'grid',
+                  gridTemplateColumns:
+                    isGroupNode || isLocationNode ? 'auto minmax(0, 1fr)' : 'minmax(0, 1fr)',
+                  gap: '6px',
+                }}
               >
-                <ExpandMore style={{cursor: 'pointer'}}>
-                  <Icon name="expand_more" color={Colors.Gray500} />
-                </ExpandMore>
-              </Popover>
+                {isGroupNode ? <Icon name="asset_group" /> : null}
+                {isLocationNode ? <Icon name="folder_open" /> : null}
+                <MiddleTruncate text={displayName} />
+              </div>
+              {isAssetNode ? (
+                <div
```

Comment:
It feels like there are enough "isGroupNode" type of stuff in here to justify three separate `Row` components. I wonder if we could invert this a bit and make a common RowContainer and then pass in the important bits like the icon and popover menu.